src/functionComponent.js

```diff
@@ -1,142 +1,134 @@
-            if (copyTodo[i].description === input.description) {
-                copyTodo[i].status = !copyTodo[i].status
-            }
-        }
-        setList({
-            ...list, toDoListItems: copyTodo
-        })
-    }
+  const removeTask = (index) => {
+    const todoCopy = [...list.toDoListItems]
+    todoCopy.splice(index, 1)
+    setList({
+      toDoListItems: todoCopy
+    })
+  }
 
-    const removeTask = (index) => {
-        const todoCopy = [...list.toDoListItems]
-        todoCopy.splice(index, 1)
-        setList({
-          ...list, toDoListItems: todoCopy
-        })
-      }
-    
-    useEffect(() => {
-        fetch('http://localhost:3004/items')
-            .then(resp => {
-                return resp.json()
-            })
-            .then(resp => {
-                setList({ ...list, toDoListItems: resp })
-            })
-    }
-    )
-    
-    return (
-        <>
-        <div
-        style={{
-          textAlign: "center",
-          maxWidth: "950px",
-          margin: "0 auto",
-          border: "1px solid #e6e6e6",
-          padding: "40px 25px",
-          marginTop: "50px"
-        }}
-      >
-        <h1> Todo List</h1>
-        <h2>
-          <form id="newText">
-            <input
-              type="text"
-              placeholder="Enter task..."
-              value={list.toDoDescription}
-              onChange={handleChange}
-            />
-            <button
-              type="submit"
-              onClick={(e) => {
-                e.preventDefault()
-                setList({
-                  ...list, toDoListItems: [...list.toDoListItems, { status: false, description: entry.taskInput }]
-                })
-                fetch('http://localhost:3004/items', {
-                  method: 'POST',
-                  headers: { "Content-Type": "application/json" },
-                  body: JSON.stringify({ status: false, description: entry.taskInput })
-                }).then(() => {
-                  console.log('task added to DB')
-                })
-              }
-              }
-            >
-              +
-            </button>
-          </form>
-        </h2>
-        <h2>
-          <ol>
-            {
-              list.toDoListItems.map((input, index) => {
-                return (
-                  <div>
-                    <li
-                      onClick={
-                        () => {
-                          completeTask(input)
-                        }
-                      }
-                    >
-                      {input.description}
-                      <input
-                        type="checkbox"
-                        checked={input.status}
-                      />
-                    </li>
+  useEffect(() => {
+    fetch('http://localhost:3005/items')
+      .then(resp => {
+        return resp.json()
+      })
+      .then(resp => {
+        setList(resp)
+      })
+  }, [])
```

Comment:
This is the line that killed us :joy: 

We needed to add an empty array to only run `#fetch` on the first render. [Check this link for more details](https://www.w3schools.com/react/react_useeffect.asp).

I assume this would also fix the issue we saw when using context & `#useReducer` but maybe we'll save that for your next project?